### PR TITLE
New MiscUtils.h functions: random_index, vector_get_random, capitalize_string_words, grab_token_string_pos

### PR DESCRIPTION
**Pull request by Bumber64**
Add new functions to `MiscUtils.h`.

`random_index` and `vector_get_random` are for getting a random entry from a vector. Common use case  is to fill up a temp vector with possible results (e.g., raws matching conditions) then need to choose one at random.

`capitalize_string_words` and `grab_token_string_pos` are from `g_src/basics.cpp` and have been tidied up to return strings (instead of modifying the existing string in the case of `capitalize_string_words`.)

Moved `random_int` up in `MiscUtils.h` so I could use it in `random_index`. (Trying to implement `random_index` in `MiscUtils.cpp` is a nightmare of cryptic errors related to `#include` that I've given up on.)

### CODE REVIEW

**Review by myk002 — COMMENTED** [review]
Do you plan to do a pass through the code and refactor to use these? If we're adding library code, I'd like to see a plan for using it.

**Comment by Bumber64** [comment]
> Do you plan to do a pass through the code and refactor to use these? If we're adding library code, I'd like to see a plan for using it.

The code is based off of stuff used by v50.11 announcement-related functions, so I'm getting all the pieces ready now. (E.g., choose random valid alert sound, capitalize a unit's translated name/title, parse markup_text_boxst tokens.)

There are probably places in existing code where we could use this stuff, too.

File: docs/dev/Lua API.rst [hunk 1]
```diff
@@ -941,6 +941,10 @@ can be omitted.
   Note that the returned string may be longer than the input string. For
   example, ``ä`` is replaced with ``a``, and ``æ`` is replaced with ``ae``.
 
+* ``dfhack.capitalizeStringWords(string)``
+
+  Return a version of the string with each word capitalized.
```
[the remark this answers is not in the document]
Comment (Bumber64) [reply docs/dev/Lua API.rst:946]: It doesn't down-case. We have a `toLower(string)` function available, though.

[the remark this answers is not in the document]
Comment (myk002) [reply docs/dev/Lua API.rst:946]: there's a native Lua string:lower() which probably makes it unnecessary

**Review by myk002 — APPROVED** [review]

File: docs/dev/Lua API.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bumber64) [reply docs/dev/Lua API.rst:946]: Do we want a CP437 version?

[the remark this answers is not in the document]
Comment (myk002) [reply docs/dev/Lua API.rst:946]: oh, that's a good point. yes, that might be useful.

**Pull request merged**